drivers/SmartThings/matter-evse/src/init.lua

```diff
@@ -0,0 +1,628 @@
+local capabilities = require "st.capabilities"
+local MatterDriver = require "st.matter.driver"
+local clusters = require "st.matter.clusters"
+local log = require "log"
+local utils = require "st.utils"
+local matter_driver_template = {}
+local embedded_cluster_utils = require "embedded_cluster_utils"
+
+local version = require "version"
+
+-- Include driver-side definitions when lua libs api version is < 11
+if version.api < 11 then
+  clusters.EnergyEvse = require "EnergyEvse"
+  clusters.ElectricalPowerMeasurement = require "ElectricalPowerMeasurement"
+  clusters.ElectricalEnergyMeasurement = require "ElectricalEnergyMeasurement"
+  clusters.DeviceEnergyManagementMode = require "DeviceEnergyManagementMode"
+  clusters.EnergyEvseMode = require "EnergyEvseMode"
+end
+
+local COMPONENT_TO_ENDPOINT_MAP = "__component_to_endpoint_map"
+local SUPPORTED_EVSE_MODES_MAP = "__supported_evse_modes_map"
+local SUPPORTED_DEVICE_ENERGY_MANAGEMENT_MODES_MAP = "__supported_device_energy_management_modes_map"
+local RECURRING_REPORT_POLL_TIMER = "__recurring_report_poll_timer"
+local RECURRING_POLL_TIMER = "__recurring_poll_timer"
+local LAST_REPORTED_TIME = "__last_reported_time"
+local TOTAL_CUMULATIVE_ENERGY_IMPORTED =
+"__total_cumulative_energy_imported" -- total in case there are multiple electrical sensors
+
+local TIMER_REPEAT = (1 * 60)        -- 1 minute
+local REPORT_TIMEOUT = (15 * 60)     -- Report the value each 15 minutes
+local MAX_CHARGING_CURRENT_CONSTRAINT = 80000 -- In v1.3 release of stack, this check for 80 A is performed.
```

Comment:
I implemented a method in electrical sensor for using either the 15 minute report interval OR a longer one, if the device sends reports at longer intervals. It may be useful to implement that here as well?

Reply:
We can use the same method in electrical sensor.